File: app/src/main/java/in/completecourse/questionbank/adapter/SpinnerAdapterType.java
package in.completecourse.questionbank.adapter;

import android.content.Context;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import in.completecourse.questionbank.R;

public class SpinnerAdapterType extends ArrayAdapter<String> {
    private final LayoutInflater mLayoutInflater;
    private final String[] mStringArray;
    private final int mResource;

    public SpinnerAdapterType(@NonNull Context context, int resource, String[] strArr) {
        super(context, resource, 0, strArr);
        this.mLayoutInflater = LayoutInflater.from(context);
        this.mResource = resource;
        this.mStringArray = strArr;
    }

    private View homeView(int i, ViewGroup viewGroup){
        View inflate = this.mLayoutInflater.inflate(this.mResource, viewGroup, false);
        ((TextView)inflate.findViewById(R.id.offer_type_txt)).setText(Html.fromHtml(this.mStringArray[i]));
        return inflate;
    }

    @Override
    public View getDropDownView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if (convertView == null){
            convertView = this.mLayoutInflater.inflate(R.layout.spinner_new_view, parent, false);
        }
        ((TextView) convertView.findViewById(R.id.offer_type_txt)).setText(Html.fromHtml(this.mStringArray[position]));
        return convertView;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        return homeView(position, parent);
    }
}

File: app/src/main/java/in/completecourse/questionbank/adapter/SpinnerAdapterType.kt
package `in`.completecourse.questionbank.adapter

import `in`.completecourse.questionbank.R
import android.content.Context
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class SpinnerAdapterType(context: Context, resource: Int, strArr: Array<String>) : ArrayAdapter<String?>(context, resource, 0, strArr) {
    private val mLayoutInflater: LayoutInflater = LayoutInflater.from(context)
    private val mStringArray: Array<String> = strArr
    private val mResource: Int = resource
    private fun homeView(i: Int, viewGroup: ViewGroup): View {
        val inflate = mLayoutInflater.inflate(mResource, viewGroup, false)
        (inflate.findViewById<View>(R.id.offer_type_txt) as TextView).text = Html.fromHtml(mStringArray[i])
        return inflate
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        var myView = convertView
        if (myView == null) {
            myView = mLayoutInflater.inflate(R.layout.spinner_new_view, parent, false)
        }
        (myView!!.findViewById<View>(R.id.offer_type_txt) as TextView).text = Html.fromHtml(mStringArray[position])
        return myView
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return homeView(position, parent)
    }

}